contracts/src/models/stamina.cairo

```diff
@@ -62,9 +61,15 @@ impl StaminaCustomImpl of StaminaCustomTrait {
 
         assert(maxes.len() > 0, 'No troops in army');
 
-        self.amount = maxes.min().unwrap();
-        self.last_refill_tick = armies_tick_config.current();
+        maxes.min().unwrap()
+    }
+
+    fn refill(ref self: Stamina, world: IWorldDispatcher) {
+        self.amount = self.max(world);
         self.sset(world);
```

Comment:
what's this function? Can we give a more explicit name to it please?

Reply:
lol you wrote it. refill stamina. maybe reset?